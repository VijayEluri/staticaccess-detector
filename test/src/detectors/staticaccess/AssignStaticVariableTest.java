package detectors.staticaccess;

/**
 * Assigning a static variable test.
 * @author Adam Warski (adam at warski dot org)
 */
public class AssignStaticVariableTest {
	private static String x;

	@StaticIndependent
	public static void m1() {
	    x = "a";		// error
	}

	public static void m2() {
		x = "b";
	}
}
