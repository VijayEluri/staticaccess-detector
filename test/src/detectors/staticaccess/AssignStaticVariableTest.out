H C SANA: Static-independent method detectors.staticaccess.AssignStaticVariableTest.m1() accesses static variable detectors.staticaccess.AssignStaticVariableTest.x  At AssignStaticVariableTest.java:[line 12]